src/components/ScanPatientPage/BarcodeScan.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useState } from "react";
+import { useHistory } from "react-router-dom";
+import Quagga from "quagga";
+import { useQuery } from "react-apollo";
+import { FETCH_ALL_PATIENTS } from "../../graphql/queries/patients";
+
+import "../../styles/BarcodeScan.css";
+
+const BarcodeScan = ({
+  eventID,
+  ccpID,
+}: {
+  eventID: string;
+  ccpID: string;
+}) => {
+  const history = useHistory();
+  const { data, loading } = useQuery(FETCH_ALL_PATIENTS);
+  const [barcode, setBarcode] = useState<string>("");
+
+  useEffect(() => {
+    if (!loading && barcode !== "") {
+      const selectedPatient = data.patients.filter(
+        (patient) => patient.barcodeValue.toString() === barcode
+      );
+
+      if (selectedPatient.length > 0) {
+        // Found patient
+        const {
```

Comment:
Correct me if I am wrong on this Jason, but are you unpacking the patient data to get the CCP ID over here depending on whether you can match a patient in our database to an existing barcode value?

Reply:
yeah so I'm getting the CCP ID so I can redirect to the proper edit screen for the existing barcode